code/modules/mob/living/carbon/human/species.dm

```diff
@@ -1204,13 +1204,19 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 
 		var/obj/item/bodypart/affecting = target.get_bodypart(ran_zone(user.zone_selected))
 
-		if(!damage || !affecting)
+		var/miss_chance = 100//calculate the odds that a punch misses entirely. considers stamina and brute damage of the puncher. punches miss by default to prevent weird cases
+		if(atk_verb == ATTACK_EFFECT_KICK && user.dna.species.punchdamagelow != 0)//kicks never miss (provided your species deals more than 0 damage)
```

Comment:
```suggestion
		if(atk_verb == ATTACK_EFFECT_KICK && user.dna.species.punchdamagelow)//kicks never miss (provided your species deals more than 0 damage)
```

Reply:
You're checking punchdamagelow multiple times, what about putting a singular check in the front so you can ensure no bad math?

Reply:
Will have to look at this later